### DISCUSSION TITLE
Is it possible to save where the usb fan controllers live?

### DISCUSSION THREAD

**Discussion by roblouie** [devcompl/TTController#103] [Q&A] [answered]
I have 2 fan controllers (due to having 6 fans). Due to TT's great competence and creating hardware, half the time I boot my PC up only one of the controllers is recognized by the app. Also only one is being recognized as an LEDFanBox under "Other devices" in windows Bluetooth & other devices screen.

However, even when it doesn't show up in TTController or Windows, both controllers show up as "Generic USB Hub" in Device Manager.  My current fix for my issue has been to uninstall the non-working one from device manager, then doing "scan for hardware changes". Then I can restart the server and I'm good.

However, I was wondering, since the devices are always plugged in, and always recognized by windows...would it be possible to put the port or location in the json file.

Idea being the service doesn't have to scan each time, if I can hardcode the addresses of them or whatever, you just read/write to that? Rather than scanning. I'm a web dev not a hardware dev so honestly I'm not sure if that is a thing you can really do, but if you can, would that solve my issue? And just let you always communicate with the controllers?

**Comment by MoshiMoshi0** [comment]
I think port identifiers are exactly what you are describing. The format is `[<vendor id>, <product id>, <port>]`, vendor id and product id are used to point to a specific usb device since the combination of the two should be unique.
It shouldn't matter where they are or how they are named in device manager, mine is "USB Input Device" for some reason.

If you get this issue again, try running the debug report, it should show 2 entries under "HID" section, but only one under "Controllers" section.

**Comment by roblouie** [comment]
I did check the debug when it didn't show up, only one item was under HID section as well. It's very possible that makes it impossible for you to do anything about it. It's really thermaltake being thermaltake and being completely incompetent.

I applaud you efforts at making them usable, but I just ordered some corsair fans. I'm going to hope microcenter lets me return the TT ones even though I don't have the box.

**Comment by MoshiMoshi0** [comment, marked as answer]
Yea, if it doesnt show in the HID section I can't do much. I bet its some issue with the controllers when they are daisy chained, maybe also linked to some motherboards.
If you want to play with it some more I would try different usb hub on the motherboard, or connecting each controller to a seperate hub if you have the cables. Or even using simple micro usb cables and connecting both controllers to separate external usb ports.
